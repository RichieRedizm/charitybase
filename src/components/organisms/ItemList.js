import React from 'react'
import PropTypes from 'prop-types'
import Item from './Item'

const ItemList = ({ charities, loading }) => {
  if (loading || charities === null) {
    return 'Loading....'
  }
  if (charities.length === 0) {
    return 'Sorry, there are no charities that match that criteria.'
  }
  return (
    <div data-testid='charities-list'>
      {charities.map(charity => (
        <Item key={charity.id} charity={charity} />
      ))}
    </div>
  )
}

ItemList.contextTypes = {
  charities: PropTypes.array,
  loading: PropTypes.bool
}

export default ItemList
