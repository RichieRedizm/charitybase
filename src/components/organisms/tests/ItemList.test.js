import React from 'react'
import { render, cleanup } from '@testing-library/react'
import ItemList from '../ItemList'

afterEach(cleanup)

test('if charities context is empty do not display any list', () => {
  const props = {
    charities: [],
    loading: false
  }
  const { queryByTestId } = render(<ItemList {...props} />)
  expect(queryByTestId('charities-list')).toBeNull()
})

test('if charities data available generate list of items', () => {
  const props = {
    charities: [
      {
        id: '266065',
        names: [
          {
            value: 'Wandsworth Mind',
            primary: true
          }
        ],
        contact: {
          email: 'finance@bwwmind.org.uk',
          person: '',
          phone: '02072598100',
          address: [
            'Brent Wandsworth & Westminster Mind',
            '6 Osbert Street',
            'London'
          ],
          postcode: 'SW1P 2QU'
        },
        finances: [
          {
            income: 0,
            spending: 0
          }
        ],
        grants: null
      },
      {
        id: '517267',
        names: [
          {
            value: 'Carmarthen Mind Association',
            primary: true
          }
        ],
        contact: {
          email: 'carmsmind132@aol.com',
          person: '',
          phone: '01267222990',
          address: ['Mind Carmarthen', '132B Priory Street', 'Carmarthen'],
          postcode: 'SA31 1LR'
        },
        finances: [
          {
            income: 0,
            spending: 0
          }
        ],
        grants: [
          {
            id: '360G-LBFEW-103658',
            amountAwarded: 15100,
            fundingOrganization: [
              {
                name: 'Lloyds Bank Foundation for England and Wales'
              }
            ]
          }
        ]
      },
      {
        id: '219829',
        names: [
          {
            value: 'The Elliott Charity',
            primary: true
          }
        ],
        contact: {
          email: 'contact@mind.org.uk',
          person: '',
          phone: '02082152295',
          address: ['Mind N A M H', 'Granta House', '15-19 Broadway', 'London'],
          postcode: 'E15 4BQ'
        },
        finances: [
          {
            income: 0,
            spending: 0
          }
        ],
        grants: null
      }
    ],
    loading: false,
    searchCharities: jest.fn()
  }
  const { getByTestId, asFragment } = render(<ItemList {...props} />)
  const charityList = getByTestId('charities-list')
  expect(charityList)
  expect(charityList.children.length).toBe(3)
  expect(asFragment()).toMatchSnapshot()
})
