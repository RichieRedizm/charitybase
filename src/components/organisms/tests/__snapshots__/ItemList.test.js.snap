// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if charities data available generate list of items 1`] = `
<DocumentFragment>
  <div
    data-testid="charities-list"
  >
    <div
      class="card grid-item"
    >
      <div>
        <h4>
          Wandsworth Mind
        </h4>
        <div>
          <strong
            class="label-dark"
          >
            Email: 
          </strong>
          finance@bwwmind.org.uk
        </div>
        <div
          class="address-wrap"
        >
          <strong
            class="label-dark"
          >
            Address:
          </strong>
          <br />
          <div>
            Brent Wandsworth & Westminster Mind,
          </div>
          <div>
            6 Osbert Street,
          </div>
          <div>
            London,
          </div>
          <div>
            SW1P 2QU
          </div>
        </div>
      </div>
      <div>
        <strong
          class="label-dark"
        >
          Finances:
        </strong>
        <div>
          <strong
            class="label-light"
          >
            Total Income: 
          </strong>
          £0
        </div>
        <div>
          <strong
            class="label-light"
          >
            Total Spending: 
          </strong>
          £0
        </div>
      </div>
      <div>
        <strong
          class="label-dark"
        >
          No Grant
        </strong>
      </div>
    </div>
    <div
      class="card grid-item"
    >
      <div>
        <h4>
          Carmarthen Mind Association
        </h4>
        <div>
          <strong
            class="label-dark"
          >
            Email: 
          </strong>
          carmsmind132@aol.com
        </div>
        <div
          class="address-wrap"
        >
          <strong
            class="label-dark"
          >
            Address:
          </strong>
          <br />
          <div>
            Mind Carmarthen,
          </div>
          <div>
            132B Priory Street,
          </div>
          <div>
            Carmarthen,
          </div>
          <div>
            SA31 1LR
          </div>
        </div>
      </div>
      <div>
        <strong
          class="label-dark"
        >
          Finances:
        </strong>
        <div>
          <strong
            class="label-light"
          >
            Total Income: 
          </strong>
          £0
        </div>
        <div>
          <strong
            class="label-light"
          >
            Total Spending: 
          </strong>
          £0
        </div>
      </div>
      <div>
        <strong
          class="label-dark"
        >
          Grants Awarded: 
        </strong>
        <div>
          <strong
            class="label-light"
          >
            Lloyds Bank Foundation for England and Wales: 
          </strong>
          £15100
        </div>
        <strong
          class="label-dark"
        >
          Total Grant: 
        </strong>
        £15100
      </div>
    </div>
    <div
      class="card grid-item"
    >
      <div>
        <h4>
          The Elliott Charity
        </h4>
        <div>
          <strong
            class="label-dark"
          >
            Email: 
          </strong>
          contact@mind.org.uk
        </div>
        <div
          class="address-wrap"
        >
          <strong
            class="label-dark"
          >
            Address:
          </strong>
          <br />
          <div>
            Mind N A M H,
          </div>
          <div>
            Granta House,
          </div>
          <div>
            15-19 Broadway,
          </div>
          <div>
            London,
          </div>
          <div>
            E15 4BQ
          </div>
        </div>
      </div>
      <div>
        <strong
          class="label-dark"
        >
          Finances:
        </strong>
        <div>
          <strong
            class="label-light"
          >
            Total Income: 
          </strong>
          £0
        </div>
        <div>
          <strong
            class="label-light"
          >
            Total Spending: 
          </strong>
          £0
        </div>
      </div>
      <div>
        <strong
          class="label-dark"
        >
          No Grant
        </strong>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
